src/main/java/teammates/ui/controller/InstructorStudentListPageData.java

```diff
@@ -1,22 +1,86 @@
 import teammates.common.datatransfer.CourseAttributes;
 import teammates.common.datatransfer.InstructorAttributes;
+import teammates.common.util.Assumption;
+import teammates.common.util.Const;
+import teammates.ui.template.InstructorStudentsListFilterBox;
+import teammates.ui.template.InstructorStudentsListFilterCourses;
+import teammates.ui.template.InstructorStudentsListSearchBox;
+import teammates.ui.template.InstructorStudentsListStudentsTableCourses;
 
 public class InstructorStudentListPageData extends PageData {
 
-    public InstructorStudentListPageData(AccountAttributes account) {
+    private InstructorStudentsListSearchBox searchBox;
+    private InstructorStudentsListFilterBox filterBox;
+    private List<InstructorStudentsListStudentsTableCourses> studentsTable;
+    private int numOfCourses;
+    public List<CourseAttributes> courses;
+    public Boolean displayArchive;
+    public String searchKey;
+    
+    public InstructorStudentListPageData(AccountAttributes account, String searchKey,
+                                         boolean displayArchive,
+                                         Map<String, InstructorAttributes> instructors,
+                                         Map<String, String> numStudents,
+                                         List<CourseAttributes> courses) {
         super(account);
+        this.courses = courses;
+        this.displayArchive = displayArchive;
+        this.searchKey = searchKey;
+        init(account, searchKey, displayArchive, instructors, numStudents, courses);
     }
+    
+    // data.isCourseArchived, course.id, data.account.googleId, data.numStudents.get(course.id),
+    // data.getInstructorCourseEnrollLink(course.id), course.name
 
-    public HashMap<String, InstructorAttributes> instructors;
-    public HashMap<String, String> numStudents;
-    public List<CourseAttributes> courses;
-    public String searchKey;
-    public Boolean displayArchive;
+    private void init(AccountAttributes account, String searchKey, boolean displayArchive,
+                      Map<String, InstructorAttributes> instructors, Map<String, String> numStudents,
+                      List<CourseAttributes> courses) {
+        this.searchBox = new InstructorStudentsListSearchBox(getInstructorSearchLink(), searchKey, account.googleId);
+        List<InstructorStudentsListFilterCourses> coursesForFilter =
+                                        new ArrayList<InstructorStudentsListFilterCourses>();
+        List<InstructorStudentsListStudentsTableCourses> coursesForStudentsTable =
+                                        new ArrayList<InstructorStudentsListStudentsTableCourses>();
+        for (CourseAttributes course : courses) {
+            InstructorAttributes instructor = instructors.get(course.id);
+            boolean isCourseArchived = isCourseArchived(course.id, instructor.googleId);
+            boolean isCourseDisplayed = displayArchive || !isCourseArchived;
```

Comment:
If you swap the ordering, you can short-circuit the expression and skip checking if it's archived when the archived ones should be displayed.